### ISSUE TITLE
[help needed] Command line arguments

### ISSUE THREAD

**Issue by zalun** [mozilla/chromeless#118]
I'd like to use chromeless to create a few psudo applications for the website I run quite often (evernote, jsfiddle, ...).

Instead of creating a new app for every URL, I'd prefer to make a URL a command line argument and change the iframe src in JavaScript.
`/path/to/chromeless . example.com`

**Comment by lloyd** [comment]
what platform?  could environment variables be used for this?

**Comment by zalun** [comment]
Linux
Environment variables would work for this use case
I could write a shell script which would change the arguments to environment variable and use them inside of the app, however it sounds a bit strange and will work for Un*x like systems only

**Comment by lloyd** [comment]
We the problem with introducing command line options that change the behavior of the app is that end users won't be consuming stuff in this fashion.  End users will get a dmg, or an installer, or a directory (linux) with a binary in it that they can run or double click to execute their application.

Given the runtime environment of apps on osx and windows, we'd need to figure out something cross platform.

Here's a proposal, how about we allow arbitrary application specific content in a key in appinfo.json?  That way you can parameterize your apps with that?

Here's a rundown of appinfo.json today: http://mozilla.github.com/chromeless/#guide/startup-parameters

We could add an 'app' key there that you can specify and could be whatever you want.

**Comment by zalun** [comment]
Will it then appify it so, the program will become a separate application? I think it would be fine

**Comment by lloyd** [comment]
yes!  it would become a separate app.  I'll add this in the next couple days for you with a little bit of sample code (how to access the configuration parameters from your app code)

**Comment by keithamus** [comment]
I have been trying to get the nsICommandLine api working in Chromeless, but it doesn't seem to be working so well.

I managed to get a environment variables module working with some simple code:

```
const {Cc,Ci} = require("chrome");
exports.get = function(aName) {
  return Cc["@mozilla.org/process/environment;1"].createInstance(Ci.nsIEnvironment).get(aName);
};
exports.set = function(aName, aValue) {
  return Cc["@mozilla.org/process/environment;1"].createInstance(Ci.nsIEnvironment).set(aName, aValue);
};
exports.exists = function(aName) {
  return Cc["@mozilla.org/process/environment;1"].createInstance(Ci.nsIEnvironment).exists(aName);
};
```

This seems to work fine but I cannot do the same thing with nsICommandLine. I would like this API to be exposed in the same way.

tl;dr: +1 for command line arguments.
